py-polars/tests/unit/test_series.py

```diff
@@ -2310,3 +2310,20 @@ class XSeries(pl.Series):
 def test_builtin_abs() -> None:
     s = pl.Series("s", [-1, 0, 1, None])
     assert abs(s).to_list() == [1, 0, 1, None]
+
+
+@pytest.mark.parametrize(
+    "value, unit, exp",
+    [
+        (13285, "d", date(2006, 5, 17)),
+        (1147880044, "s", datetime(2006, 5, 17, 15, 34, 4)),
+        (1147880044 * 1_000, "ms", datetime(2006, 5, 17, 15, 34, 4)),
+        (1147880044 * 1_000_000, "us", datetime(2006, 5, 17, 15, 34, 4)),
+        (1147880044 * 1_000_000_000, "ns", datetime(2006, 5, 17, 15, 34, 4)),
+    ],
+)
+def test_from_epoch_expr(value: int, unit: str, exp: date | datetime) -> None:
```

Comment:
I think you need to make `unit` here an `EpochTimeUnit`, right now mypy sees a generic string, and cannot figure out that it is valid `EpochTimeUnit`.